### PR TITLE
Update cmash to 0.5.0

### PR DESCRIPTION
**Pull request by BiocondaBot**
Update [`cmash`](https://bioconda.github.io/recipes/cmash/README.html): **0.4.0** &rarr; **0.5.0**

[install with bioconda](http://bioconda.github.io/recipes/cmash/README.html) [Conda](https://anaconda.org/bioconda/cmash/files)

Info | Link
-----|-----
Recipe | [`recipes/cmash`](https://github.com//bioconda/bioconda-recipes/tree/bump/cmash/recipes/cmash) (click to view/edit other files)
Releases |[https://github.com/dkoslicki/CMash/tags](https://github.com/dkoslicki/CMash/tags)
Recipe Maintainer(s) | @luizirber, @dkoslicki
Author | @dkoslicki
***

This pull request was automatically generated (see [docs](https://bioconda.github.io/contributor/updating.html)).

### CODE REVIEW

**Comment by luizirber** [comment]
Since CMash is using type hints, it will have to constrain Python to `python >= 3.5` (build failed on Python 2.7)

**Comment by luizirber** [comment]
Oh, and f-strings, so `python >= 3.6`

**Comment by luizirber** [comment]
@dkoslicki you can pull this branch locally and push a new commit to add the constraint.

```bash
$ git clone git@github.com:bioconda/bioconda-recipes.git -b bump/cmash
$ cd bioconda-recipes
<... edit recipes/cmash/meta.yaml ...>
$ git commit && git push
```
should work

**Comment by dkoslicki** [comment]
@luizirber I will work on this locally and see if I can update it and get the bioconda testing environment setup as well.

File: recipes/cmash/meta.yaml [hunk 1]
```diff
@@ -15,7 +15,7 @@ build:
 
 requirements:
   host:
-    - python
+    - python >= 3.6
```
Comment (luizirber) [inline recipes/cmash/meta.yaml]: No space here:
```suggestion
    - python >=3.6
```

**Comment by luizirber** [comment]
Hi @bioconda/python, seems like we are hitting a similar issue to https://github.com/bioconda/bioconda-recipes/pull/21101#issuecomment-604099844 here.  It's failing with
```
00:32:38 BIOCONDA INFO (OUT) + MakeStreamingDNADatabase.py -h

00:32:38 BIOCONDA INFO (OUT) /opt/conda/conda-bld/cmash_1585182579362/test_tmp/run_test.sh: /opt/conda/conda-bld/cmash_1585182579362/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_pl/bin/MakeStreamingDNADatabase.py: /opt/conda/conda-bld/cmash_1585182579362/_h_env_placehold_placehold_placehold_: bad interpreter: No such file or directory
```
The original `#!` is `#! /usr/bin/env python` in https://github.com/dkoslicki/CMash/blob/b34d4c5ff1a75becbdd3bbda66fbb31874bf5cec/scripts/MakeStreamingDNADatabase.py#L1

Any pointers?

**Comment by mbargull** [comment]
@luizirber, thanks for the ping. We updated to `conda-build=3.19.1` 2 days ago to patch a bug we encountered. But that version apparently introduced another one. I cherry-picked the patch we needed in https://github.com/conda-forge/conda-build-feedstock/pull/131 .

I restarted the CI and it should work again now.

**Comment by mbargull** [comment]
> The original `#!` is `#! /usr/bin/env python` ...

If you wondered how this got replaced, this was `pip` replacing it with the build (`host`) environment-specific one, for which `conda-build` then records the path. When the built package is installed by `conda`, `conda` can replace the recorded path with the final environment's one. The "`conda-build` then records" part didn't work with `conda-build=3.19.1`.

**Comment by dkoslicki** [comment]
Thanks @mbargull! Glad to know it wasn’t something on my end. In that case (FYI @luizirber), I am going to undo some of the changes I made to the `meta.yaml` file. In particular, I introduced more strict requirements (never a good thing if not strictly required) and stripped a few things out that I would like to put back in). After that, I’ll add the “please review & merge tag”

**Comment by dpryan79** [comment]
@bioconda-bot please merge

**Pull request merged**